app/Classes/MediaHandler/VideoHandler.php

```diff
@@ -105,4 +101,32 @@ public function getVersions(Media $media): array
             'versions' => $versions->pluck('created_at', 'number')->map(fn($date) => strtotime($date)),
         ];
     }
+
+    /**
+     * @return array
+     */
+    public function purgeDerivatives(): array
+    {
+        $failedMediaIds = [];
+
+        foreach (Media::whereType(MediaType::VIDEO)->get() as $media) {
+            // Restore latest version to (re-)generate derivatives.
+            $version = $media->latestVersion;
+
+            $oldVersionNumber = $version->number;
+            $wasProcessed = $version->processed;
+
+            $version->update(['number' => $media->latestVersion->number + 1, 'processed' => 0]);
```

Comment:
this shouldnt be run twice at the same time

Reply:
This will most likely not run twice at the same time, and even if, there should be no problems besides some lost computing time.